### ISSUE TITLE
Not resized when contents is deleted

### ISSUE THREAD

**Issue by adam-lynch** [jackmoore/autosize#238]
I'm updating from 1.x.x to the latest. It resizing when shift+return is pressed but not if I select all the text and press del. Any ideas? 

I use flexbox if that helps. I tried stripping down my app into a simple codepen but it's hard. I probably stripped my styles down too much because I couldn't reproduce it.

I've only tested in IE11 and Chrome 43 for Windows. 

I'm digging into the code but it's hard to tell what *should* be happening. If it helps, I've noticed that on [this line](https://github.com/jackmoore/autosize/blob/master/dist/autosize.js#L83) in my app, `endHeight` is always the same as `startHeight` and `originalHeight`.

**Comment by adam-lynch** [comment]
Ok I've got a better idea of what's going on now.

We've complicated styles. We previously set the height of the `form` wrapping the `textarea` in the callack called by autosize. When this callback is / isn't called or what it does to the field while calculating the new height is different somehow in a way that breaks our convoluted usage. 

What we used to do was get the new height of the field in the callback, check it versus the "previous" height (which we've cached in a variable) and if it's different, set the height of the form to `auto`, then get the height of the field and set the height of the form appropriately.

What I think is happening now: I'm catching delete and backspace keyDowns myself, then immediately setting the `form`'s height to `auto`, then autosize sets the height of the field to `auto`, then calculates and sets it again, then my callback gets called by autosize and sets the height of the `form` appropriately.

Aaanyway, I'll close this.
